homeassistant/components/template/switch.py

```diff
@@ -141,12 +142,19 @@ def template_switch_state_listener(entity, old_state, new_state):
         @callback
         def template_switch_startup(event):
             """Update template on startup."""
-            async_track_state_change(
-                self.hass, self._entities, template_switch_state_listener
-            )
+            if self._template is not None:
+                async_track_state_change(
+                    self.hass, self._entities, template_switch_state_listener
+                )
 
             self.async_schedule_update_ha_state(True)
 
+        # restore state after startup
+        await super().async_added_to_hass()
+        state = await self.async_get_last_state()
+        if state:
+            self._state = state.state in ("true", STATE_ON)
```

Comment:
State will always be `STATE_ON`, it can never be `true`. 
```suggestion
            self._state = state.state == STATE_ON
```